src/App.jsx

```diff
@@ -14,33 +17,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortedBy, setSortedBy] = useState(null);
+  const [reversed, setReversed] = useState(false);
+
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
+
+    if (reversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSortedBy('alphabet');
+    setReversed(false); // Reset reversed state after sorting
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    if (reversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSortedBy('length');
+    setReversed(false); // Reset reversed state after sorting
+  };
+
+  const reverseOrder = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+    setReversed(!reversed);
+  };
+
+  const resetOrder = () => {
+    setGoods(goodsFromServer);
+    setSortedBy(null);
+    setReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortedBy === 'alphabet' && !reversed ? '' : 'is-light'}`}
```

Comment:
pls use [classnames](https://www.npmjs.com/package/classnames) library for such cases
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**